packages/zod/src/__tests__/contract.type_spec.ts

```diff
@@ -0,0 +1,9 @@
+import { expectType } from 'tsd';
+import { Contract } from '@farfetched/core';
+import { string } from 'zod';
+
+import { zodContract } from '../zod_contract';
+
+const contract = zodContract(string());
+
+expectType<Contract<unknown, string>>(contract);
```

Comment:
Same here, I would like to have tests for type inference in more complex cases like records, arrays, optional fields.